skpro/distributions/cmp_pois_gam.py

```diff
@@ -0,0 +1,171 @@
+        2D np.ndarray, same shape as ``self``
+            pdf values at the given points
+        """
+        from scipy.special import gamma as gam_fun
+
+        lam = self.lambda_
+        alpha = self.alpha
+        beta = self.beta
+        pdf_value = np.zeros_like(x)
+        tol = 1e-10
+
+        for idx, val in np.ndenumerate(x):
+            if val <= 0:
+                continue  # PDF is zero for non-positive values
+
+            const_term = np.exp(-beta * val) / ((np.exp(lam) - 1) * val)
+            i = 1
+            while True:
+                t1 = lam * (pow(beta * val, alpha))
+                numer = pow(t1, i)
+                i_fact = math.factorial(i)
+                gamma_fun = gam_fun(i * alpha)
+                denom = i_fact * gamma_fun
+
+                term = numer / denom
+                pdf_value[idx] += term
+
+                if term < tol:
+                    break
+                i += 1
+                if i > 1000:  # safeguard to prevent infinite loop
+                    break
+            pdf_value[idx] = pdf_value[idx] * const_term
+
+        return pdf_value
+
+    def _pmf(self, k):
+        """Probability mass function.
+
+        Parameters
+        ----------
+        k : 2D np.ndarray, same shape as ``self``
+            values to evaluate the pmf at
+
+        Returns
+        -------
+        2D np.ndarray, same shape as ``self``
+            pmf values at the given points
+        """
+        from scipy.stats import poisson
+
+        lambda_ = self.lambda_
+        return poisson.pmf(k, lambda_)
+
+    def _compute_crj(self, r, j, rho):
+        from itertools import combinations_with_replacement
+
+        from scipy.special import comb
+
+        partitions = np.array(
+            [
+                p
+                for p in combinations_with_replacement(range(1, r + 1), j)
+                if sum(p) == r
+            ]
+        )
+        if partitions.size == 0:
+            return 0
+
+        term = np.prod([comb(rho + 1 + s_i, s_i + 2) for s_i in partitions.T], axis=0)
+        c_rj = np.sum(term)
+        c_rj *= (rho**2 + rho) ** (-r / 2 - j)
+
+        return c_rj
+
+    def _compute_hr(self, x, r, rho):
+        from scipy.special import eval_hermitenorm, factorial
+
+        hr = np.zeros_like(x, dtype=float)
+        for j in range(1, r + 1):
+            H = eval_hermitenorm(r + 2 * j, x)
+            crj = self._compute_crj(r, j, rho)
+            hr += H * crj / factorial(j)
+        return hr
+
+    def _cdf(self, x):
+        """Cumulative distribution function.
+
+        Parameters
+        ----------
+        x : 2D np.ndarray, same shape as ``self``
+            values to evaluate the cdf at
+
+        Returns
+        -------
+        2D np.ndarray, same shape as ``self``
+            cdf values at the given points
+        """
+        rho = self.alpha
+        max_r = 10  # Adjust as needed
+        phi_x = np.exp(-(x**2) / 2) / np.sqrt(2 * np.pi)
```

Comment:
> Could you please point out where it seems to be exactly the pdf or cdf of a normal distribution.

normal pdf

Reply:
Ah I see. I shall replace that with the normal pdf.